node/forest_libp2p/src/service.rs

```diff
@@ -187,13 +187,14 @@ where
                                 response,
                             }).await;
                         }
-                        ForestBehaviourEvent::BlockSyncRequest { channel, .. } => {
-                            // TODO implement blocksync provider
-                            let _ = channel.send(BlockSyncResponse {
-                                chain: vec![],
-                                status: 203,
-                                message: "handling requests not implemented".to_owned(),
-                            }).await;
+                        ForestBehaviourEvent::BlockSyncRequest { channel, peer, request } => {
+                            debug!("Received blocksync request (peerId: {:?})", peer);
```

Comment:
@austinabell you've suggested to `thread::spawn` here? I thought about it, but In order to do that, the `Arc<DB>` will need to be changed to `Arc<RwLock<DB>>` I think, however DB is already being passed around in several services with just `Arc`

Reply:
yeah, I was attempting `RwLock`, because it implements `Sync + Send`, `task::spawn` was complaining that `DB` is not thread-safe. 
Solved it now by simply adding `+Sync+Send` to `DB`.

Reply:
yeah just to clarify on that, `task::spawn_blocking` should be used, as the database access and this logic is synchronous